crypto/keys/keybase_base.go

```diff
@@ -59,9 +88,14 @@ func newBaseKeybase(optionsFns ...KeybaseOption) baseKeybase {
 	return baseKeybase{options: options}
 }
 
-// baseSecpPrivKeyGen generates a secp256k1 private key from the given bytes
-func baseSecpPrivKeyGen(bz [32]byte) tmcrypto.PrivKey {
-	return secp256k1.PrivKeySecp256k1(bz)
+// StdPrivKeyGen generates a secp256k1 private key from the given bytes
+func StdPrivKeyGen(bz []byte, algo SigningAlgo) tmcrypto.PrivKey {
```

Comment:
This seems...hacky. Why do we need this? For testing purposes only? The method name is not indicative of it only supporting secp256k1 keys. Can we at least expand it to support all other known key types too?

Reply:
Well, by default, I think we only want to support secp256k1, as that's what the current `x/auth` module in the Cosmos SDK does.  But I agree, it's probably a good ideas to add `SecpPrivKeyGen` as separate function, and have `StdPrivKeyGen` use it.  That was when we add other keys as the Std defaults, we just add them to the `Std` function.

Reply:
Do we not want the standard default to be only supporting secp keys?